centinel/primitives/http.py

```diff
@@ -49,10 +52,64 @@ def get_request(host, path="/", headers=None, ssl=False,
     }
     # the external result is used when threading to store
     # the results in the list container provided.
-    if external is not None and type(external) is dict:
-        external[url] = result
+    # if external is not None and type(external) is dict:
+    #    external[url] = result
     return result
 
+
+def get_request(host, path="/", headers=None, ssl=False,
+                external=None, url=None):
+    http_results = {}
+    first_response = _get_http_request(host, path, headers, ssl, external, url)
+    if "failure" in first_response["response"]:  # If there was an error, just ignore redirects and return
+        if external is not None and type(external) is dict:
+            external[url] = first_response
+            return first_response
+    # Checks HTTP Status code and location header to see if the webpage calls for a redirect
+    is_redirecting = str(first_response["response"]["status"]).startswith("3") and "location"\
+                                            in first_response["response"]["headers"]
+    if is_redirecting:
+        http_results["request"] = first_response["request"]
+        http_results["redirects"] = {}
+        http_results["redirects"]["0 (" + host + path + ")"] = first_response["response"]
+        redirect_result = None
+        redirect_number = 1
+        while redirect_result is None or (str(redirect_result["response"]["status"]).startswith("3") and
```

Comment:
There is a lot of text happening here and I would prefer for a) the addition of some helper variables, e.g. compute a boolean called is_redirect at the end of the loop and b) adding parens to make this section more readable.